pkg/model/multiversion/manager.go

```diff
@@ -35,31 +36,40 @@ var (
 // of each non-deprecated version with their correspending ackmodel.Model
 // and APIInfos.
 type APIVersionManager struct {
-	gitRepo *git.Repository
+	gitRepo  *git.Repository
+	metadata ackmetadata.ServiceMetadata
 
 	hubVersion    string
 	spokeVersions []string
 
-	deprecatedVersions []string
-	removedVersions    []string
-
-	apiInfos map[string]APIInfo
+	apiInfos map[string]ackmetadata.APIInfo
 	models   map[string]*ackmodel.Model
 }
 
 // NewAPIVersionManager initialises and returns a new APIVersionManager.
 func NewAPIVersionManager(
 	sdkCacheDir string,
+	metadataPath string,
 	serviceAlias string,
-	hubVersion string,
-	apisInfo map[string]APIInfo,
+	apisInfo map[string]ackmetadata.APIInfo,
 	defaultConfig ackgenconfig.Config,
 ) (*APIVersionManager, error) {
 	if len(apisInfo) == 0 {
 		return nil, fmt.Errorf("empty apisInfo")
 	}
 
-	spokeVersions := make([]string, 0, len(apisInfo)-1)
+	metadata, err := ackmetadata.NewServiceMetadata(metadataPath)
+	if err != nil {
+		return nil, err
+	}
+
+	hubVersion, err := metadata.GetLatestAPIVersion()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
The hub version is not always the latest version. I would like to keep the `APIVersionManager` flexible in term of choosing a hub version. However we can still enforce "choosing the latest version as hub" in the build script or `cmd` / `pkg/generate/ack`

Reply:
Sounds good. Will wait for your changes to `cmd` - we can load it in there.